pkg/controller/daemon/daemon_controller.go

```diff
@@ -904,31 +905,53 @@ func (dsc *DaemonSetsController) syncNodes(ds *extensions.DaemonSet, podsToDelet
 
 	glog.V(4).Infof("Nodes needing daemon pods for daemon set %s: %+v, creating %d", ds.Name, nodesNeedingDaemonPods, createDiff)
 	createWait := sync.WaitGroup{}
-	createWait.Add(createDiff)
 	template := util.CreatePodTemplate(ds.Spec.Template, ds.Spec.TemplateGeneration, hash)
-	for i := 0; i < createDiff; i++ {
-		go func(ix int) {
-			defer createWait.Done()
-			err := dsc.podControl.CreatePodsOnNode(nodesNeedingDaemonPods[ix], ds.Namespace, &template, ds, metav1.NewControllerRef(ds, controllerKind))
-			if err != nil && errors.IsTimeout(err) {
-				// Pod is created but its initialization has timed out.
-				// If the initialization is successful eventually, the
-				// controller will observe the creation via the informer.
-				// If the initialization fails, or if the pod keeps
-				// uninitialized for a long time, the informer will not
-				// receive any update, and the controller will create a new
-				// pod when the expectation expires.
-				return
-			}
-			if err != nil {
-				glog.V(2).Infof("Failed creation, decrementing expectations for set %q/%q", ds.Namespace, ds.Name)
+	// Batch the pod creates. Batch sizes start at 1 and double with each
+	// successful iteration in a kind of "slow start". This handles attempts
+	// to start large numbers of pods that would likely all fail with the
+	// same error. For example a project with a low quota that attempts to
+	// create a large number of pods will be prevented from spamming the
+	// API service with the pod create requests after one of its pods fails.
+	// Conveniently, this also prevents the event spam that those failures
+	// would generate.
+	for batchSize, pos := 1, 0; createDiff > pos; batchSize, pos = integer.IntMin(2*batchSize, createDiff-(pos+batchSize)), pos+batchSize {
```

Comment:
can you move the initial batch size to a constant that all controllers reference?

that way we can tweak it globally from 1, 2, 4, 8, ... in one place.